trollflow2/launcher.py

```diff
@@ -319,24 +328,27 @@ def get_dask_client(config):
 
 def queue_logged_process(msg, prod_list, produced_files, log_queue):
     """Run `process` with a queued log."""
-    _reset_log_handlers()
-    handler = handlers.QueueHandler(log_queue)
-    LOG.addHandler(handler)
+    setup_queued_logging(log_queue)
+    with suppress(ValueError):
+        signal.signal(signal.SIGUSR1, print_traces)
+        LOG.debug("Use SIGUSR1 on pid {} to check the current tracebacks of this subprocess.".format(os.getpid()))
```

Comment:
Should the log message be before the traces are printed? It'll look weird in the logs in this order.

Reply:
the `signal.signal` call just sets up the call back to run when the signal happens, so this won't print any traces at that moment.
The reason why the log message happens after is that if that if that function is not called in the main thread of a process, it fails, and thus the log message shouldn't be printed.